lib/phoenix_live_view/helpers.ex

```diff
@@ -370,6 +370,101 @@ defmodule Phoenix.LiveView.Helpers do
     EEx.compile_string(expr, options)
   end
 
+  @doc """
+  Returns the entry errors for an upload.
+
+  The following errors may be returned:
+
+    * `:too_large` - The entry exceeded the `:max_file_size` constraint
+    * `:too_many_files` - The number of selected files exceeds the `:max_entries` constraint
+    * `:not_acceptable` - The entry does not match the `:accept` MIME types
+
+  ## Examples
+
+      def error_to_string(:too_large), do: "Too large"
+      def error_to_string(:too_many_files), do: "You have selected too many files"
+
+      <%= for entry <- @uploads.avatar.entries do %>
+        <%= for err <- upload_errors(@uploads.avatar, entry) do %>
+          <div class="alert alert-danger">
+            <%= error_to_string(err) %>
+          </div>
+        <% end %>
+      <% end %>
+  """
+  def upload_errors(
+        %Phoenix.LiveView.UploadConfig{} = conf,
+        %Phoenix.LiveView.UploadEntry{} = entry
+      ) do
+    for {ref, error} <- conf.errors, ref == entry.ref, do: error
+  end
+
+  @doc """
+  Generates an image preview on the client for a selected file.
+
+  ## Examples
+
+      <%= for entry <- @uploads.avatar.entries do %>
+        <%= live_img_preview entry, width: 75 %>
+      <% end %>
+  """
+  def live_img_preview(%Phoenix.LiveView.UploadEntry{ref: ref} = entry, opts \\ []) do
+    binding_prefix = opts[:binding_prefix] || "phx-"
+
+    opts =
+      Keyword.merge(opts,
+        id: "phx-preview-#{ref}",
+        data_phx_upload_ref: entry.upload_ref,
+        data_phx_entry_ref: ref
+      ) ++ [{"#{binding_prefix}hook", "Phoenix.LiveImgPreview"}]
+
+    assigns = %{opts: opts}
+
+    ~L"""
+    <%= Phoenix.HTML.Tag.content_tag(:img, "", @opts) %>
+    """
```

Comment:
I believe there is no benefit in wrapping this in `~L`:

```suggestion
    Phoenix.HTML.Tag.content_tag(:img, "", @opts)
```